workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/EntityGameObjectLinker.cs

```diff
@@ -102,16 +102,22 @@ public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, ViewCom
         ///     Unlinks a GameObject and ECS Entity.
         /// </summary>
         /// <remarks>
-        ///    The GameObject and ECS Entity should be linked before this method call.
+        ///     The GameObject and ECS Entity should be linked before this method call.
         /// </remarks>
         /// <param name="gameObject">The GameObject to unlink.</param>
         /// <param name="entity">The ECS Entity to unlink.</param>
         /// <param name="viewCommandBuffer">
         ///     An instance of the ViewCommandBuffer. Must be flushed to apply the changes to the ECS Entity.
         /// </param>
-        public void UnlinkGameObjectFromEntity(GameObject gameObject, Entity entity, ViewCommandBuffer viewCommandBuffer)
+        public void UnlinkGameObjectFromEntity(GameObject gameObject, Entity entity,
+            ViewCommandBuffer viewCommandBuffer)
         {
-            if (entityManager.Exists(entity))
+            if (!entityManager.Exists(entity))
+            {
+                return;
```

Comment:
The approach in this PR feels more like a hack. This if check here for instance feels like a place where we'd throw errors if we wanted to do things "properly". I'm ok with merging this PR if the idea is to quickly patch things up for the short term. Ideally, we should ensure that all gameobjects are cleaned up before unity's automatic scene cleanup behaviour kicks in. This was achieved in the SDK by cleaning up all gameobjects in an OnApplicationQuit method attached to the bootstrap gameobject representing the worker connection. That method was fired before unity's own scene cleanup.

Reply:
Yeah this is a shorter term fix for a crash on close in the ECS-FPS client

Reply:
Ok, that makes sense. Can you create a ticket for cleaning this up properly?

Reply:
Yes will do